sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationLiveTests.cs

```diff
@@ -41,132 +41,142 @@ public DocumentModelAdministrationLiveTests(bool isAsync, DocumentAnalysisClient
         public async Task BuildDocumentModelCanAuthenticateWithTokenCredential()
         {
             var client = CreateDocumentModelAdministrationClient(useTokenCredential: true);
-            var trainingFilesUri = new Uri(TestEnvironment.BlobContainerSasUrl);
-
-            var modelId = Recording.GenerateId();
-            BuildDocumentModelOperation operation = await client.BuildDocumentModelAsync(WaitUntil.Completed, trainingFilesUri, DocumentBuildMode.Template, modelId);
-
-            // Sanity check to make sure we got an actual response back from the service.
-            Assert.IsNotNull(operation.Value.ModelId);
-
-            await client.DeleteDocumentModelAsync(modelId);
-        }
-
-        [RecordedTest]
-        [TestCase(true)]
-        [TestCase(false)]
-        public async Task BuildDocumentModel(bool singlePage)
-        {
-            var client = CreateDocumentModelAdministrationClient();
-            var trainingFilesUri = new Uri(singlePage ? TestEnvironment.BlobContainerSasUrl : TestEnvironment.MultipageBlobContainerSasUrl);
             var modelId = Recording.GenerateId();
+            var trainingFilesUri = new Uri(TestEnvironment.BlobContainerSasUrl);
 
-            BuildDocumentModelOperation operation = await client.BuildDocumentModelAsync(WaitUntil.Completed, trainingFilesUri, DocumentBuildMode.Template, modelId);
-
-            Assert.IsTrue(operation.HasValue);
-
-            DocumentModelDetails model = operation.Value;
-
-            ValidateDocumentModelDetails(model);
-
-            Assert.AreEqual(1, model.DocumentTypes.Count);
-            Assert.IsTrue(model.DocumentTypes.ContainsKey(modelId));
+            BuildDocumentModelOperation operation = null;
 
-            DocumentTypeDetails documentType = model.DocumentTypes[modelId];
+            try
+            {
+                operation = await client.BuildDocumentModelAsync(WaitUntil.Completed, trainingFilesUri, DocumentBuildMode.Template, modelId);
+            }
+            finally
+            {
+                if (operation != null && operation.HasValue)
+                {
+                    await client.DeleteDocumentModelAsync(modelId);
+                }
+            }
 
-            Assert.AreEqual(DocumentBuildMode.Template, documentType.BuildMode);
+            // Sanity check to make sure we got an actual response back from the service.
+            Assert.AreEqual(modelId, operation.Value.ModelId);
         }
 
         [RecordedTest]
-        public async Task BuildDocumentModelWithNeuralBuildMode()
+        [TestCaseSource(nameof(s_buildDocumentModelTestCases))]
+        public async Task BuildDocumentModel(DocumentBuildMode buildMode)
```

Comment:
Merging `BuildDocumentModel` (with Template mode) and `BuildDocumentModelWithNeuralBuildMode` in a single test.